test/functional/formatting_stack/mount.clj

```diff
@@ -0,0 +1,60 @@
+(ns functional.formatting-stack.mount
+  (:require
+   [clojure.test :refer [deftest is testing]]
+   [formatting-stack.mount :as sut]
+   [formatting-stack.reporters.passthrough :as reporters.passthrough]
+   [mount.core :as mount :refer [defstate]]
+   [nedap.utils.spec.api :refer [check!]]))
+
+(def sample-report
+  {:source   ::some-linter
+   :level    :warning
+   :column   40
+   :line     1
+   :msg      "omg"
+   :filename "foo.clj"})
+
+(defn proof [a]
+  (reify formatting-stack.protocols.linter/Linter
+    (--lint! [this filenames]
+      (check! empty? filenames)
+      (reset! a [sample-report]))))
+
+(defstate stateful-component
+  :start 42
+  :stop 0)
+
+(deftest works
+  (testing "It can be started/stopped without errors"
+    (let [p (atom nil)
+          ;; pass an empty stack (except for the `proof`), so that no side-effects will be triggered (would muddy the test suite):
+          opts {:strategies               []
+                :third-party-indent-specs {}
+                :formatters               []
+                :linters                  [(proof p)]
+                :processors               []
+                :in-background?           false
+                :reporter                 (reporters.passthrough/new)}]
+      (sut/configure opts)
+      (mount/start)
```

Comment:
It would help me to understand - can `(mount/start)` perform code-loading (like a `clojure.core/require` or a `tools.namespace.repl/refresh`)?

If that were the case, this test would be unlike its Component and Integrant cousins

Reply:
`mount/start` isn't really smart in-and-of-itself, it really just calls `:start` on all the `defstates` that it knows about.

However, `mount` automatically handles re-compilation of its `defstates` whenever a namespace is recompiled.

So if someone ran `(require ... :reload)` or ran `C-c C-k` in a CIDER repl on a namespace with a `defstate`, all hooks would be called.